src/rpc/rpc.hpp

```diff
@@ -43,8 +45,9 @@ class c_rpc_server final {
 		std::thread m_thread;
 		Mutex m_session_vector_mutex;
 		std::list<c_session> m_session_list;
-		std::map<std::string, std::function<std::string(const std::string)>> m_rpc_functions_map;
+		std::map<std::string, std::function<nlohmann::json(const std::string)>> m_rpc_functions_map;
 		std::array<unsigned char, crypto_auth_hmacsha512_KEYBYTES> m_hmac_key; ///< for hmac authentication key, shold be loaded from conf file TODO
+		xint m_session_counter = 0;
```

Comment:
This variable can be use by multiple threads if we will add them. Should be GUARDED_BY(m_session_vector_mutex)

Reply:
I will fix mulithread bugs